sentry-core/src/main/java/io/sentry/core/SendCachedEventFireAndForgetIntegration.java

```diff
@@ -35,6 +35,15 @@ public void register(@NotNull IHub hub, @NotNull SentryOptions options) {
       ExecutorService es = Executors.newSingleThreadExecutor();
       es.submit(
           () -> {
+            if (!hub.isIntegrationEnabled(SendCachedEventFireAndForgetIntegration.class)) {
```

Comment:
As the other comment. I know we've discussed using the type as a possibility but it has gotchas. I believe should use reference comparison to make sure it's the one instance added.

Reply:
answer: https://github.com/getsentry/sentry-android/pull/245#discussion_r370108607

Reply:
Will resolve this since we'll settle on the behavior on the other comment.